h2o-algos/src/main/java/hex/tree/DTree.java

```diff
@@ -600,9 +600,17 @@ public DecidedNode(UndecidedNode n, DHistogram hs[], Constraints cs) {
         // which might be because all predictor columns are now constant, or
         // because all responses are now constant.
         _splat = Float.NaN;
-        Arrays.fill(_nids,ScoreBuildHistogram.UNDECIDED_CHILD_NODE_ID);
+        Arrays.fill(_nids, ScoreBuildHistogram.UNDECIDED_CHILD_NODE_ID);
         return;
       }
+      if(cs != null) {
```

Comment:
it could be useful to have an option to disable the check - for almost all our self-checks we have a mechanism to defeat them using a system property

Reply:
Good point! Is it ok now? https://github.com/h2oai/h2o-3/pull/5837/commits/5fe68994f90632c1700a5c19783cbfaae332a080